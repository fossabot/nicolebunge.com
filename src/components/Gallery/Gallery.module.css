.gallery {
  display: grid;
  grid-auto-flow: dense;
  grid-gap: 1.5em;
  grid-template-columns: repeat(12, 1fr);
  list-style: none;
  margin: 0;
  padding: 0;
}

.gallery__item:nth-child(1) {
  grid-column-end: span 6;
  grid-column-start: 1;
}

.gallery__item:nth-child(2) {
  grid-column-end: span 5;
  grid-column-start: 8;
}

.gallery__item:nth-child(3) {
  grid-column-end: span 5;
  grid-column-start: 2;
}

.gallery__item:nth-child(4) {
  grid-column-end: span 4;
  grid-column-start: 9;
}

.gallery__item:nth-child(5) {
  align-self: flex-end;
  grid-column-end: span 4;
  grid-column-start: 1;
}

.gallery__item:nth-child(6) {
  align-self: flex-end;
  grid-column-end: span 7;
  grid-column-start: 6;
}
