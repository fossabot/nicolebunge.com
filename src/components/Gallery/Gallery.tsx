/* eslint-disable camelcase */

import clsx from 'clsx';
import React from 'react';
import CloudinaryImage from '../CloudinaryImage';
import { gallery, gallery__item } from './Gallery.module.css';

const sizes = [
  {
    width: 336,
    height: 224,
  },
  {
    width: 459,
    height: 573,
  },
  {
    width: 336,
    height: 420,
  },
  {
    width: 213,
    height: 213,
  },
  {
    width: 581,
    height: 465,
  },
  {
    width: 459,
    height: 573,
  },
];

interface Image {
  id: string;
  public_id: string;
}

interface GalleryProps {
  className?: string;
  images: Image[];
}

function Gallery(props: GalleryProps): JSX.Element {
  const { className, images, ...otherProps } = props;

  return (
    <ul className={clsx(className, gallery)} {...otherProps}>
      {images.map((image, index) => (
        <li className={gallery__item} key={image.id}>
          <figure>
            <CloudinaryImage {...image} {...sizes[index % sizes.length]} />
          </figure>
        </li>
      ))}
    </ul>
  );
}

export type { GalleryProps };
export default Gallery;
