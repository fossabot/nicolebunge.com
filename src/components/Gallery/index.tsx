import classNames from 'classnames';
import React, { FC } from 'react';
import CloudinaryImage, { CloudinaryImageProps } from '../CloudinaryImage';
import * as styles from './style.module.css';

export interface GalleryProps {
  className?: string;
  images: CloudinaryImageProps[];
}

const Gallery: FC<GalleryProps> = (props) => {
  const { className, images, ...otherProps } = props;

  return (
    <ul className={classNames(className, styles.gallery)} {...otherProps}>
      {images.map((image) => (
        <li className={styles.gallery__item} key={image.src}>
          <figure>
            <CloudinaryImage {...image} />
          </figure>
        </li>
      ))}
    </ul>
  );
};

export default Gallery;
