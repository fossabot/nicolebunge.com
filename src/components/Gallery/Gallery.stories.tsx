import Gallery from '.';

export default {
  component: Gallery,
};

export const Default = {
  args: {
    images: [
      {
        src: 'path/to/image.jpg',
        alt: 'Image alt tag',
      },
    ],
  },
};
