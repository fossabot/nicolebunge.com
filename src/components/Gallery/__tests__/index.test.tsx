/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Gallery from '..';

describe('Gallery component', () => {
  it('renders correctly', () => {
    const images = [
      {
        src: 'path/to/image.jpg',
        alt: 'Image alt tag',
      },
    ];
    const { container } = render(<Gallery images={images} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
