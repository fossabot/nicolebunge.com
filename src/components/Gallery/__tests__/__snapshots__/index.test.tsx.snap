// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component renders correctly 1`] = `
<ul
  class=""
>
  <li>
    <figure>
      <img
        alt="Image alt tag"
        loading="lazy"
        src="https://res.cloudinary.com/nicolebunge/image/upload/f_auto,q_auto/v1627923793/nicolebunge.com/path/to/image.jpg"
      />
    </figure>
  </li>
</ul>
`;
